packages/compass-generative-ai/src/components/ai-optin-modal.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+  Body,
+  Link,
+  MarketingModal,
+  SpinLoader,
+  css,
+  spacing,
+  useDarkMode,
+} from '@mongodb-js/compass-components';
+import { AISignInImageBanner } from './ai-signin-banner-image';
+import { closeOptInModal, optIn } from '../store/atlas-optin-reducer';
+import type { RootState } from '../store/atlas-ai-store';
+
+const GEN_AI_FAQ_LINK = 'https://www.mongodb.com/docs/generative-ai-faq/';
+
+type OptInModalProps = {
+  isOptInModalVisible: boolean;
+  isOptInInProgress: boolean;
+  onOptInModalClose: () => void;
+  onOptInClick: () => void;
+};
+
+const titleStyles = css({
+  marginBottom: spacing[400],
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+});
+
+const disclaimer = css({
+  padding: `0 ${spacing[900]}px`,
+});
+
+const AIOptInModal: React.FunctionComponent<OptInModalProps> = ({
+  isOptInModalVisible,
+  isOptInInProgress,
+  onOptInModalClose,
+  onOptInClick,
+}) => {
+  const darkMode = useDarkMode();
+
+  return (
+    <MarketingModal
+      darkMode={darkMode}
```

Comment:
We shouldn't need to pass darkMode here, it'll inherit it from the context and the theme mode provider in the react tree. For some more context, here's where that provider is in the react tree:
https://github.com/mongodb-js/compass/blob/206341cf72ac9e81670400fd41235577e67c833d/packages/compass-web/src/entrypoint.tsx#L312

Reply:
resolved!